src/tools/illink/src/linker/Linker.Steps/ValidateVirtualMethodAnnotationsStep.cs

```diff
@@ -51,8 +51,8 @@ void ValidateMethodRequiresUnreferencedCodeAreSame (OverrideInformation ov)
 					nameof (RequiresUnreferencedCodeAttribute),
 					method.GetDisplayName (),
 					baseMethod.GetDisplayName ());
-				IMemberDefinition origin = (ov.IsOverrideOfInterfaceMember && ov.InterfaceImplementor.Implementor != method.DeclaringType)
-					? ov.InterfaceImplementor.Implementor
+				IMemberDefinition origin = (ov.IsOverrideOfInterfaceMember && ov.RuntimeInterfaceImplementation.Implementor != method.DeclaringType)
+					? ov.RuntimeInterfaceImplementation.Implementor
 					: method;
 				Context.LogWarning (origin, DiagnosticId.RequiresUnreferencedCodeAttributeMismatch, message);
```

Comment:
It looks like this is the reason for the `IL2046` warning not being suppressed.
```C#
class Base : IFoo
{
  [Ruc("Message")]
  [UnconditionalSuppressMessage("IL2046")]
  void M() { } // Suppressed IL2046
}

class Derived : Base, IFoo // Another 2046 (unsuppressed)
{
}

interface IFoo
{
  void M();
}
```
Since `BaseType` provides the interface implementation method for `Derived`, we decided to put the warning method on the type that implements the interface. I think that makes sense for if `Derived` implements `IFoo` and `Base` doesn't, but if both do, it might make sense to report it on the implementation method.